static/js-src/upcoming-page.js

```diff
@@ -30,69 +32,51 @@ async function init() {
     features[channel] = await featuresPromise[channel];
   }
 
+
   // Remove the loading sign once the data has been fetched from the APIs
   document.body.classList.remove('loading');
 
+  const upcomingEl = document.querySelector('chromedash-upcoming');
   channelsArray.forEach((channel) => {
-    channels[channel].components = mapFeaturesToShippingType(features[channel]);
+    channels[channel].features = upcomingEl.mapFeaturesToShippingType(features[channel]);
   });
 
-  const upcomingEl = document.querySelector('chromedash-upcoming');
+
   upcomingEl.channels = channels;
+  upcomingEl.lastFetchedOn = channels[channelsArray[1]].version;
+  let cardsDisplayed = upcomingEl.computeItems();
+  upcomingEl.lastMilestoneVisible = channels[channelsArray[cardsDisplayed-1]].version;
+  cardsToFetchInAdvance = upcomingEl.cardsToFetchInAdvance;
 
   window.csClient.getStars().then((starredFeatureIds) => {
     upcomingEl.starredFeatures = new Set(starredFeatureIds);
   });
 }
 
-function mapFeaturesToShippingType(features) {
-  const featuresMappedToShippingType = {};
-  features.forEach(f => {
-    const component = f.browsers.chrome.status.text;
-    if (!featuresMappedToShippingType[component]) {
-      featuresMappedToShippingType[component] = [];
-    }
-    featuresMappedToShippingType[component].push(f);
-  });
-
-  for (let [, feautreList] of Object.entries(featuresMappedToShippingType)) {
-    sortFeaturesByName(feautreList);
-  }
-
-  return featuresMappedToShippingType;
-}
-
-
-/**
- *  @param {!Array<!Object>} features
- */
-function sortFeaturesByName(features) {
-  features.sort((a, b) => {
-    a = a.name.toLowerCase();
-    b = b.name.toLowerCase();
-    if (a < b) {
-      return -1;
-    }
-    if (a > b) {
-      return 1;
-    }
-    return 0;
-  });
-}
-
 // Slide to newer or older version
 function move(e) {
   let container = document.querySelector('chromedash-upcoming');
   let divWidth = container.shadowRoot.querySelector('chromedash-upcoming-milestone-card').cardWidth;
   let margin = 8;
   let change = divWidth+margin*2;
 
-  if (e.target.id=='next-button') {
-    jumpSlideWidth-= change; // move to newer version
-    container.style.marginLeft=jumpSlideWidth + 'px';
-  } else {
-    jumpSlideWidth+=change; // move to older version
-    container.style.marginLeft=jumpSlideWidth + 'px';
+  if (container.lastFetchedOn) {
+    if (e.target.id=='next-button') {
```

Comment:
Style: use spaces around operators.

Reply:
Thank you for pointing this out. 
Done. See [this](https://github.com/GoogleChrome/chromium-dashboard/pull/1458/commits/8ef87954ddf5fbf7bf9daf5d8aa238c9c074dbdd) commit.